src/engines/aurora/gui.h

```diff
@@ -114,6 +114,8 @@ class GUI : boost::noncopyable {
 	virtual void callbackRun();
 	/** Callback that's triggered when a widget was activated. */
 	virtual void callbackActive(Widget &widget);
+	/** Callback that's triggered when a text input is received */
+	virtual void callbackTextInput(Common::UString text);
```

Comment:
Again, this should probably be a const reference. The callback shouldn't usually need to modify the text input at all.